test/fail_compilation/ice14929.d

```diff
@@ -75,7 +75,7 @@ template isComponentStorage(CS, C)
     {
         CS cs = CS.init;
         ulong eid;
-        cs.add(eid, c);
+        cs.add(eid, C.init);
```

Comment:
Why was this change necessary?

Reply:
Fun reason: Previously, DMD would try to semantically evaluate `cs.add` _before_ figuring out what if anything `c` is, hit the `!in` instantiation error, and bail with the `TEST_OUTPUT` message. WIth this change, it verifies the arguments to cs.add() _first_ and immediately notices that there is no such variable as `c`, causing the `typeof` to fail and, as is `typeof`'s wont, silently return false.

Reply:
Ah, yes, of course. I was reading the corrected code and tried to figure out what was wrong with it :smiley:.

Reply:
It is a bit weird and inconsistent that `typeof` errors out if only the error that happens in `typeof` is in a different template.

But that's neither what this bug nor the testcase's bug is about, so ... :man_shrugging: